src/components/scripture/scripture.module.ts

```diff
@@ -1,7 +1,137 @@
 import { Module } from '@nestjs/common';
+import { books } from './books';
+import { ScriptureReference } from './dto';
 import { ScriptureReferenceResolver } from './scripture-reference.resolver';
 
+interface Range<T> {
+  start: T;
+  end: T;
+}
+
+type VerseRange = Range<number>;
+
+type ScriptureRange = Range<Required<ScriptureReference>>;
```

Comment:
This already exists as a DTO, it can be used instead of defining this.

Reply:
Were you just referring to ScriptureRange here? I just used a new interface for VerseRange.